libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreAlterTest.java

```diff
@@ -230,6 +230,53 @@ public void testAlterSoupWithFullTextIndexesToGetIndexesOnCreatedAndLastModified
         tryAlterSoupToGetIndexesOnCreatedAndLastModified(SmartStore.Type.full_text);
     }
 
+    /**
+     * Create soup with fts4 virtual table
+     * Call alterSoup passing in same index specs
+     * Make sure virtual table is recreated with fts5
+     * That way soup created before 4.2 (using fts4 virutal table) can be migrated to fts5 by calling alterSoup
+     * @throws JSONException
+     */
+    public void testAlterSoupwithFullTextIndexesFromFts4ToFts5() throws JSONException {
+        IndexSpec[] indexSpecs = new IndexSpec[] {new IndexSpec(CITY, SmartStore.Type.full_text), new IndexSpec(COUNTRY, SmartStore.Type.full_text)};
```

Comment:
This test demonstrates how one can use alterSoup an existing soup from using fts4 to using fts5